### ISSUE TITLE
[Everhour Time Tracking] ... "Failed to fetch projects" 

### ISSUE THREAD

**Issue by aha778** [raycast/extensions#14813]
### Extension

https://www.raycast.com/farisaziz12/everhour

### Raycast Version

1.83.2

### macOS Version

15.0.1 (24A348)

### Description

Both "Recent Tasks" and "Active Projects" are not working.
I'm getting this error description "Failed to fetch projects Cannot read properties of undefined (reading 'total')"

### Steps To Reproduce

1. Open "Recent Tasks" and "Active Projects"
2. And immediately you'll see the error, as it can't connect to the data
<img width="807" alt="Screenshot 2024-10-07 at 11 25 20" src="https://github.com/user-attachments/assets/c5107af7-95d3-48a1-b525-5221eb5b241a">

**Comment by xmok** [comment]
Hi @aha778,

Can you confirm that you DO have projects and tasks in your Everhour Dashboard AND have access to the API?
Would be helpful in figuring out what the problem could be.

**Comment by aha778** [comment]
Hi there,

Yes, I do have active projects and tasks, and the API access. 

— 
Best regards,
Founder & CEO
/Andrei

 <https://marketingo.xyz/>￼￼Follow me on LinkedIn <https://www.linkedin.com/in/andreiharet/>
for valuable insights on how to lead your
marketing strategy and team effectively.

On 7 Oct 2024, at 11:40, Syed M Mokaarim Bin Razi ***@***.***> wrote:


Hi @aha778 <https://github.com/aha778>,

Can you confirm that you DO have projects and tasks in your Everhour Dashboard AND have access to the API?
Would be helpful in figuring out what the problem could be.

**Comment by teisenhower** [comment]
Hello! I just started experiencing the same issue so I decided to fork the extension and do some digging. 

I found that my specific case was caused by some of the tasks being pulled in not having the expected `time` property. It looks as though all of these problematic tasks are ones that were deleted from their respective platform that feeds into Everhour; Jira, Monday, Asana, etc. 

Adding a quick conditional to check to for the existence of `time` prevented this error.

```ts
const aggregatedTasks = timeRecords.reduce((agg: { [key: string]: Task }, { time, task }: TimeRecordResp) => {
  if (!task || !task.time) return agg;
  const { id, name, projects, time: taskTime } = task;
  const { total = 0 } = taskTime;
  if (!agg[id]) {
    agg[id] = { id, name, projects, time: { total, recent: time } };
  } else {
    agg[id].time.recent += time;
  }
  return agg;
}, {});
```

I am sure you will want to implement a more thorough solution, just wanted to share my quick findings in hopes it helps 🤘🏼

**Comment by xmok** [comment]
Great stuff 👍.
It's a decent solution because the check before the one you added i.e. `if (!task...` itself was added to fix #13548.
Would you be willing to make a PR? 
Since I haven't come across the issue yet it's difficult for me to troubleshoot.

**Comment by teisenhower** [comment]
> Great stuff 👍. It's a decent solution because the check before the one you added i.e. `if (!task...` itself was added to fix #13548. Would you be willing to make a PR? Since I haven't come across the issue yet it's difficult for me to troubleshoot.

Thank you! 💪🏼 I'd be more than happy to create a PR!
